### ISSUE TITLE
Unsion with OpenSSH on Windows shows empty prompt

### ISSUE THREAD

**Issue by KillerLink** [bcpierce00/unison#794]
**Problem Description:**
When using Unison GTK with Windows provided OpenSSH (pubkey auth configured via `.ssh/config`), an empty prompt is presented after selecting the profile.
After Clicking "okay", regardless of what input i provide to the prompt (including none at all), programm continues to work as expected.

**Debugging Attempts and further Information:**
* No prompt or error is shown when using the same profile without modification in the terminal version (from windows cmd or windows powershell via `C:\Program Files (x86)\unison-v2.51.5-ocaml-4.12.0\bin\unison.exe test`.
* The problem also does not occur if i use PuTTy's plink.exe in combination with a batch script [ssh2plink](https://twiki.org/p/pub/Codev/UnisonKeySetup/ssh2plink.bat) (adapted paths and added the public key for authentication with `-i path/to/key` to the plink.exe invocation) for `sshcmd`.
* I tried different unison versions on the client (anything newer from the release page than the one i use currently is crashing, that is 2.52.x, i saw that this seems already to be known).
* I intercepted the program flow at various points with wrapperscripts on both windows client and linux server side to capture if arguments or environments differ in a distinguishable way, none to be found.
* As far as i traced the problem (by watching the linux servers logs), the prompt is shown after the ssh connection has already been established successfully.

**Unison Profile:** (.unison/test.prf)
```
# Unison preferences
label = unison-test-profile
root = C:\Users\MyWindowsUser\unison\test_unison
root = ssh://myuser@myserver/test_unison
sshargs = -C
#sshcmd=C:\Windows\System32\OpenSSH\ssh.exe
#sshcmd=C:\\opt\\ssh2plink.bat
```

**"Client" Version Info:**
```
Windows 10, 21H2, Build 19044.1889
OpenSSH_for_Windows_8.9p1, LibreSSL 3.0.2
unison version 2.51.5 (ocaml 4.12.0)
```

**"Server" Version Info:** 
```
Ubuntu 22.04
Linux ilx30 5.15.0-47-generic #51-Ubuntu SMP Thu Aug 11 07:51:15 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
OpenSSH_8.9p1 Ubuntu-3, OpenSSL 3.0.2 15 Mar 2022
unison version 2.51.5 (ocaml 4.12.0)
```

**Comment by tleedjarv** [comment]
Thank you for this extensive report.

There were some fixes in this area included in 2.52. To get a working binary, try this link: https://github.com/bcpierce00/unison/actions/runs/3037048829
If you are logged in and scroll down, you should see a long list of downloads. Pick a Windows version suitable for you and then a file with suffix `-publish`. This will get you a working development version, which is soon to be released as 2.53.0.

Please do try this version and report back results. If you still get the empty prompt then I will prepare a specific debug version for you to try.

**Comment by KillerLink** [comment]
No problem, thank you for the fast response!

For now, I picked `unison-d07c5f14.ocaml-4.14.0+mingw64c.windows-2022-publish.zip`, downloaded, unpacked, and executed the contained `bin/unison-gui.exe`, resulting in the same behaviour of an otherwise empty prompt where it does not matter what input i provide.

To avoid unintended misunderstands and additional clarity, i attach two screenshots, one of the original reported unison version and of the one just downloaded, showing the respective prompt.

Please let me know if I can provide any further information or do more specific testing. 

![eg1](https://user-images.githubusercontent.com/17423668/190334406-5c001862-b5c7-4fa7-a0bc-9c21e2729c96.PNG)
![eg2](https://user-images.githubusercontent.com/17423668/190334414-b12b4bc9-f60a-4cf5-84ec-ab00d7762930.PNG)

**Comment by tleedjarv** [comment]
I think I have found the issue and have created a potential fix. Could you please test a build from https://github.com/tleedjarv/unison/actions/runs/3058672047 (you may have to wait until CI completes before you see any downloads).

If this works well for you, do you also have the possibility to test if the password prompt (without pubkey auth) works correctly? That would be very welcome. If you can do it, please test with both OpenSSH and plink.exe.

**Comment by KillerLink** [comment]
I took `unison-81a3db75.ocaml-4.14.0+mingw64c.windows-2022-publish.zip` from the build you provided above.

So, in regard to the empty prompt i got good news, every pubkey-auth combination of gui/cmd with openssh/plink now seems to work as i'd expect it.

The bad news, now password-based authentication with plink, both cmd and gui, are broken.

Please find screenshots attached that i think provide the most usefull information about the current state ( i tried to keep my privacy blanks as information retaining as possible).

CMD+OpenSSH+Password (working):
![ut_cmd_openssh_password](https://user-images.githubusercontent.com/17423668/190511627-1fcec719-9eb6-46e6-8c0a-abe891e7df6b.PNG)

CMD+ssh2plink+Password (not working):
![ut_cmd_ssh2plink_password](https://user-images.githubusercontent.com/17423668/190511628-439ecbfc-bf5a-454d-8773-dbbd4798063c.PNG)

GUI+OpenSSH+Password (working):
![ut_gui_openssh_password](https://user-images.githubusercontent.com/17423668/190511629-26e6f4ee-a476-4944-b6a2-1c66fcc0ce1c.PNG)

GUI+ssh2plink+Password (not working):
![ut_gui_ssh2plink_password](https://user-images.githubusercontent.com/17423668/190511632-87645756-2cd8-41ab-b688-fceccd3e7eba.PNG)

**Comment by tleedjarv** [comment]
> So, in regard to the empty prompt i got good news, every pubkey-auth combination of gui/cmd with openssh/plink now seems to work as i'd expect it.

Thank you for testing. You can keep using this version if you wish. It is likely very close to what is going to be released as 2.53.0.

> The bad news, now password-based authentication with plink, both cmd and gui, are broken.

I think this may have never worked; plink is written in such a way. I will leave it be unless you yourself have used plink with password-auth successfully before and can confirm it's a regression.
